apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ReservedKeywords.kt

```diff
@@ -13,7 +13,8 @@ private val JAVA_RESERVED_WORDS = arrayOf(
 
 // Reference:
 // https://kotlinlang.org/docs/enum-classes.html#working-with-enum-constants:~:text=properties%20for%20obtaining%20its%20name%20and%20position
-private val KOTLIN_RESERVED_ENUM_VALUE_NAMES = arrayOf("name", "ordinal")
+// "header" and "impl" are added to this list because of https://youtrack.jetbrains.com/issue/KT-52315
```

Comment:
😱

Reply:
PR to handle this at the KotlinPoet level: https://github.com/square/kotlinpoet/pull/1248